spring.core/src/main/java/com/epam/springcore/task/controller/impl/TrainerController.java

```diff
@@ -0,0 +1,71 @@
+package com.epam.springcore.task.controller.impl;
+
+import com.epam.springcore.task.controller.ITrainerController;
+import com.epam.springcore.task.dto.PassUsernameDTO;
+import com.epam.springcore.task.dto.TrainerDTO;
+import com.epam.springcore.task.dto.TrainingDTO;
+import com.epam.springcore.task.facade.GymFacade;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/trainers")
```

Comment:
If you want to add base part such "/api", you could add it by configuration
https://www.baeldung.com/spring-boot-context-path